File: src/javafromzero/recursion/Table.java
package javafromzero.recursion;

public class Table {

    void printable(int num, int multiplier) {
        if (multiplier <= 10) {
            System.out.println(num * multiplier);
            printable(num, multiplier + 1);

        }

    }

    public static void main(String args[]) {
        Table table = new Table();
        table.printable(4, 1);
    }
}

File: src/kotlinfromzero/recursion/Table.kt
package kotlinfromzero.recursion

fun main() {
    printTableByRecursion(5, 1)
}

fun printTableByRecursion(num: Int, multiplier: Int) {
    if (multiplier <= 10) {
        println("$num * $multiplier = ${num * multiplier}")
        printTableByRecursion(num, multiplier + 1)
    }

}